### PR TITLE
feat: add optional base_url kwarg to OpenAIModel

### PR DESCRIPTION
**Pull request by sambarnes**
hihi :v: regarding the short exchange over on this issue:
https://github.com/pydantic/pydantic-ai/pull/215#issuecomment-2538677887

noticed that the proposed syntax wasnt actually available on OpenAIModel yet, so i figured that's the PR you meant you'd be open to?

if you'd rather not introduce the API addition rn,  i also have a purely docs change in this branch:

https://github.com/pydantic/pydantic-ai/compare/main...sambarnes:pydantic-ai:sam/openrouter-example

totally open to driving this PR home tho if this is along the lines of what ya were thinkin :)

### CODE REVIEW

**Comment by sambarnes** [comment]
docs running locally:
image

image

File: docs/install.md [hunk 1]
```diff
@@ -128,6 +128,23 @@ agent = Agent(model)
 ...
 ```
 
+#### `base_url` argument
+
+To use another OpenAI compatible API, such as [OpenRouter](https://openrouter.ai), you can configure that via the [`base_url` argument][pydantic_ai.models.openai.OpenAIModel.__init__]:
```
Comment (dmontagu) [inline docs/install.md]:
```suggestion
To use another OpenAI-compatible API, such as [OpenRouter](https://openrouter.ai), you can make use of the [`base_url` argument][pydantic_ai.models.openai.OpenAIModel.__init__]:
```

File: pydantic_ai_slim/pydantic_ai/models/openai.py [hunk 2]
```diff
@@ -75,22 +76,25 @@ def __init__(
             model_name: The name of the OpenAI model to use. List of model names available
                 [here](https://github.com/openai/openai-python/blob/v1.54.3/src/openai/types/chat_model.py#L7)
                 (Unfortunately, despite being ask to do so, OpenAI do not provide `.inv` files for their API).
+            base_url: The base url for the OpenAI requests, if not provided, the `OPENAI_BASE_URL` environment variable
```
Comment (dmontagu) [inline pydantic_ai_slim/pydantic_ai/models/openai.py]:
```suggestion
            base_url: The base url for the OpenAI requests. If not provided, the `OPENAI_BASE_URL` environment variable
```

File: pydantic_ai_slim/pydantic_ai/models/openai.py [hunk 3]
```diff
@@ -75,22 +76,25 @@ def __init__(
...
+                will be used if available. Otherwise, defaults to OpenAI's base url.
             api_key: The API key to use for authentication, if not provided, the `OPENAI_API_KEY` environment variable
                 will be used if available.
             openai_client: An existing
                 [`AsyncOpenAI`](https://github.com/openai/openai-python?tab=readme-ov-file#async-usage)
-                client to use, if provided, `api_key` and `http_client` must be `None`.
+                client to use, if provided, `base_url`, `api_key`, and `http_client` must be `None`.
```
Comment (dmontagu) [inline pydantic_ai_slim/pydantic_ai/models/openai.py]:
```suggestion
                client to use. If provided, `base_url`, `api_key`, and `http_client` must be `None`.
```

**Review by dmontagu — APPROVED** [review]
A few minor copy edits but this looks good to me.

**Comment by samuelcolvin** [comment]
thanks so much.

**Pull request merged**